node/lib/util/git_util.js

```diff
@@ -997,3 +997,29 @@ exports.getReference = co.wrap(function *(repo, name) {
     }
     return null;
 });
+
+/**
+ * Read an index file, and overwrite an in-memory index with its contents.
+ *
+ * @param {NodeGit.Index} index
+ * @param {String|undefined} path if set, whence to read the index
+ */
+exports.overwriteIndexFromFile = co.wrap(function*(index, path) {
```

Comment:
why not just open a new index and replace the old index instance?

Reply:
Because the old index instance is what repo.index() returns, and there are places that call that to get the index.